### DISCUSSION TITLE
Dynamic functions + Media Queries

### DISCUSSION THREAD

**Discussion by sergeylaptev** [jpudysz/react-native-unistyles#391] [Q&A] [answered]
Hello and first of all thanks for the great library!:) I have a question regarding complex styling where I need to combine several styling props together and also change their look starting from specific breakpoint. From what I can see in the docs right now you can either have media queries support (via "mq" import and further usage in the stylesheet object) or complex combination of several styling props using dynamic functions (and multiple if statements inside). So the question is how to get the best of both worlds and apply media queries (let's say "starting from lg") to specific combination of styling props? Thank you!

**Comment by jpudysz** [comment]
Breakpoints are designed for that case: https://www.unistyl.es/v3/references/breakpoints

eg.:

```ts
{
    container: {
         alignItems: {
              xs: 'center', // from xs to lg use 'center'
              lg: 'flex-end' // from lg and above use 'flex-end'
         }
    }
}
```

Edit:

I would avoid `if-else` in dynamic functions. 

I hope it answers your question 😄

**Comment by sergeylaptev** [comment]
I would love to avoid them too, but this is how your v2 documentation proposes to achieve compound variants;) https://v2.unistyl.es/reference/compound-variants/

**Reply by jpudysz** [reply]
Ah so you use 2.0? I was referring to 3.0. 

Compound variants are supported from 3.0. For 2.0 I had to propose some workarounds!

**Reply by sergeylaptev** [reply]
Yes, because I'm still not sure how reliable and stable "beta" version of the library is.

**Reply by jpudysz** [reply]
Beta is not production ready. We're pattching a lot of things, especially after initial release. I would go with Unistyles 3.0 from first RC.

Anyway, you can still use breakpoints on 2.0 branch. For compound, there is no other way than if-else + dynamic function

**Reply by sergeylaptev** [reply]
Yeah it looks like for v2 the best bet would be to add same "breakpoints" object into theme itself. Get current breakpoint numeric value inside dynamic function like this
```ts
const breakpoint = theme.breakpoints[rt.breakpoint]
```

and then make media queries-like comparisons manually
```ts
if (breakpoint > theme.breakpoints.lg) {
  ....
}
```

I don't like this approach a lot as it seems cursed to me, but it is what it is for v2 at least. We desperately need v3 production ready solution!:)

**Reply by jpudysz** [reply, marked as answer]
We're doing what we can to ship it. Should be available somewhere in Q1 of 2025.

**Reply by sergeylaptev** [reply]
Thank you for your dedication!:)
